app/src/main/java/org/oppia/android/app/topic/ViewPagerAdapter.kt

```diff
@@ -13,24 +13,41 @@ class ViewPagerAdapter(
   private val internalProfileId: Int,
   private val topicId: String,
   private val storyId: String,
-  private val enablePracticeTab: Boolean
+  private val enablePracticeTab: Boolean,
+  private val enableMyDownloads: Boolean
 ) : FragmentStateAdapter(fragment) {
 
-  override fun getItemCount(): Int = TopicTab.getTabCount(enablePracticeTab)
+  override fun getItemCount(): Int {
+    if (enableMyDownloads) {
+      return 1
+    } else {
+      return TopicTab.getTabCount(enablePracticeTab)
+    }
+  }
 
   override fun createFragment(position: Int): Fragment {
-    return when (TopicTab.getTabForPosition(position, enablePracticeTab)) {
-      TopicTab.INFO -> {
-        TopicInfoFragment.newInstance(internalProfileId, topicId)
-      }
-      TopicTab.LESSONS -> {
-        TopicLessonsFragment.newInstance(internalProfileId, topicId, storyId)
-      }
-      TopicTab.PRACTICE -> {
-        TopicPracticeFragment.newInstance(internalProfileId, topicId)
-      }
-      TopicTab.REVISION -> {
-        TopicRevisionFragment.newInstance(internalProfileId, topicId)
+    if (enableMyDownloads) {
```

Comment:
I'm a bit confused why these tab changes are needed. Shouldn't we be navigating to the preview fragment on the outset rather than first navigating to TopicActivity? This might be a bit of a UI architecture question (/cc @rt4914 in case you have thoughts), but I'd expect the routing location (i.e. HomeActivity) to decide whether to route to the preview or full experience based on whether the topic is downloaded (to avoid complex fragment management, or transitioning subsequent activities).

Reply:
fixed